### PR TITLE
Describe structs

### PR DESCRIPTION
**Pull request by siman**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,237 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
```
Comment (bedeho) [inline src/lib.rs]: Change copyright to Jsgenesis.

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    Uint64,
+    Int16,
+    Int32,
+    Int64,
+    Text(u32),
+    Internal(ClassId),
+    // External(ExternalProperty),
+}
+
+impl Default for PropertyType {
```
Comment (bedeho) [inline src/lib.rs:120]: Why is this needed?

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    Uint64(u64),
+    Int16(i16),
+    Int32(i32),
+    Int64(i64),
+    Text(Vec<u8>),
+    Internal(EntityId),
+    // External(ExternalPropertyType),
+}
+
+impl Default for PropertyValue {
```
Comment (bedeho) [inline src/lib.rs:142]: Why is this needed?

**Review by bedeho — CHANGES_REQUESTED** [review]
1. There are 39 warnings, could you try to reduce or eliminate those?

2. Untrack Cargo.lock, libraries should not track it, therefore also add to .gitignore
https://github.com/rust-lang/cargo/issues/315

File: src/lib.rs (hunk 2 above)
Comment (siman) [reply src/lib.rs:120]: Why do we need a `Default` for `PropertyType`?
Because `PropertType` is an enum and if we want enum to be used in Substrate RPC that it should be serializable. This means that Substrate needs a default value for enums.

**Comment by siman** [comment]
> 1. There are 39 warnings, could you try to reduce or eliminate those?
> 2. Untrack Cargo.lock, libraries should not track it, therefore also add to .gitignore
>    [rust-lang/cargo#315](https://github.com/rust-lang/cargo/issues/315)

1. Warnings will be fixed in [another PR](https://github.com/Joystream/substrate-versioned-store/pull/1), because there I implement functions and warnings of unused function arguments just disappear. 
2. Fixed: Removed Cargo.lock from repo. Added Cargo.lock to .gitignore.

File: src/lib.rs (hunk 3 above)
Comment (siman) [reply src/lib.rs:142]: Answered above to a similar question.
